### PR TITLE
Adding flask monitoring page with overview of resources

### PR DESCRIPTION
**Pull request by schlupov**

### CODE REVIEW

File: resallocserver/monitoring/app.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    pools = load_config_file("/etc/resallocserver/pools.yaml")
+    resources_list = append_resources(resources.on())
+    resources_in_use = {}
+    for resource in resources_list:
+        if not resource["ticket"]:
+            continue
+        if resource["pool"] not in resources_in_use.keys():
+            resources_in_use[resource["pool"]] = 1
+        else:
+            resources_in_use[resource["pool"]] += 1
```
Comment (FrostyX) [inline resallocserver/monitoring/app.py:22]: You can avoid this if-else logic by

```
resources_in_use.setdefault(resource["pool"], 1)
resources_in_use[resource["pool"]] += 1
```

File: resallocserver/monitoring/app.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    information = {}
+    for name, number in resources_in_use.items():
```
Comment (FrostyX) [inline resallocserver/monitoring/app.py:24]: nit: I would rename `number` to `taken`

File: rpm/resalloc.spec.tpl [hunk 3]
```diff
@@ -106,6 +106,22 @@ Requires(pre): /usr/sbin/useradd
...
 
+%package webui
+Summary:    %sum - webui part
+
+Requires:   %default_python-%srcname = %version-%release
+%if %{with python3}
+Requires: python3-flask
+%else
+Requires: python-flask
+%endif
```
Comment (praiskup) [inline rpm/resalloc.spec.tpl:117]: Nit, I think we can build the resalloc-webui package only for python3.  No need to suport this on EL7.

File: resallocwebui/static/style.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+body {
+    margin: 10px;
+}
```
Comment (praiskup) [inline resallocwebui/static/style.css:3]: Missing EOL

File: resallocserver/monitoring/app.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    return render_template('resources.html', information=information)
+
+
+def append_resources(resources):
+    resources_list = []
+    for resource in resources.all():
+        resources_list.append(
+            {'id': resource.id, 'name': resource.name, 'pool': resource.pool, 'tags': ', '.join(list(resource.tag_set)),
+             'status': resource.state, 'releases': resource.releases_counter,
+             'ticket': resource.ticket.id if resource.ticket else None})
```
[the remark this answers is not in the document]
Comment (FrostyX) [reply resallocserver/monitoring/app.py]: Thanks

> Similarly for the dict above.

I would suggest reformatting also the `information[name] = ...` dict but if you like it that way, feel free to click "Resolve conversation"

File: rpm/resalloc.spec.tpl [hunk 6]
```diff
@@ -106,6 +106,18 @@ Requires(pre): /usr/sbin/useradd
 The %name-server package provides the resalloc server, and
 some tooling for resalloc administrators.
 
+%package webui
+Summary:    %sum - webui part
+
+Requires:   %default_python-%srcname = %version-%release
```
Comment (praiskup) [inline rpm/resalloc.spec.tpl:112]: I think we need to dependon `resalloc-server`, too.

File: resallocwebui/app.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+from resallocserver.logic import QResources
+from resalloc.helpers import load_config_file
+
+app = Flask(__name__)
+
+
+@app.route('/resources')
+def resources():
+    with session_scope() as session:
+        resources = QResources(session)
```
Comment (praiskup) [inline resallocwebui/app.py:14]: FTR, I think we can keep session_scope() here for now, but long term we should start using equivalent approach to what we have in Copr (flask sqlalchemy, or similar - so we have a session scope per request).

File: resallocwebui/templates/resources.html [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Resalloc resources</title>
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.3.1/dist/css/bootstrap.min.css"
+          integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
```
Comment (praiskup) [inline resallocwebui/templates/resources.html:8]: I would prefer to depend on the bootstrap package shipped in Fedora (but can be done later, too).

**Comment by schlupov** [comment]
Screenshot from 2022-04-13 11-17-17

**Comment by schlupov** [comment]
I added dependency for `resalloc-server`

File: rpm/resalloc.spec.tpl [hunk 9]
```diff
@@ -106,6 +106,20 @@ Requires(pre): /usr/sbin/useradd
 The %name-server package provides the resalloc server, and
 some tooling for resalloc administrators.
 
+%if %{with python3}
+%package webui
+Summary:    %sum - webui part
+
+Requires:   %default_python-%srcname = %version-%release
+Requires: %name-server
+Requires: python3-flask
```
Comment (praiskup) [inline rpm/resalloc.spec.tpl:115]: Can you please conditionalize this sub-package?  I'm sure python3-flask is not installable on RHEL7 at least.

**Comment by praiskup** [comment]
Otherwise LGTM

**Pull request merged**